test/analysis/buildTypeWeights.test.ts

```diff
@@ -0,0 +1,239 @@
+                    weight: 1,
+                    fields: {},
+                },
+                User: {
+                    weight: 1,
+                    fields: {
+                        name: 0,
+                        email: 0,
+                    },
+                },
+                Movie: {
+                    weight: 1,
+                    fields: {
+                        name: 0,
+                    },
+                },
+            };
+        });
+
+        // this is only if we choose to have 'query' as its own property (seperate from object types) in the user configuration options
+        xtest('query parameter', () => {
+            const typeWeightObject = buildTypeWeightsFromSchema(schema, {
+                query: 2,
+            });
+            expectedOutput.query.weight = 2;
+
+            expect(typeWeightObject).toEqual({ expectedOutput });
+        });
+
+        test('object parameter', () => {
+            const typeWeightObject = buildTypeWeightsFromSchema(schema, {
+                object: 2,
+            });
+
+            expectedOutput.user.weight = 2;
+            expectedOutput.movie.weight = 2;
+
+            expect(typeWeightObject).toEqual({ expectedOutput });
+        });
+
+        test('scalar parameter', () => {
+            const typeWeightObject = buildTypeWeightsFromSchema(schema, {
+                scalar: 2,
+            });
+
+            expectedOutput.user.fields.name = 2;
+            expectedOutput.user.fields.email = 2;
+            expectedOutput.movie.fields.name = 2;
+
+            expect(typeWeightObject).toEqual({ expectedOutput });
+        });
+    });
+
+    describe('throws an error if...', () => {
+        beforeEach(() => {
+            schema = buildSchema(`
+                type Query {
+                    user: User
+                    movie: Movie
+                }
+                
+                type User {
+                    name: String
+                    email: String
+                }
+                
+                type Movie {
+                    name: String
+                    director: User
+                }   
+            `);
+        });
+
+        test('user configures the type weights with negative numbers', () => {
+            // check that the error thrown from the function includes the substring 'negative' to inform the user of negative problem
+            expect(() => buildTypeWeightsFromSchema(schema, { object: -1 })).toThrowError(
+                'negative'
+            );
+            expect(() => buildTypeWeightsFromSchema(schema, { mutation: -1 })).toThrowError(
+                'negative'
+            );
+            expect(() => buildTypeWeightsFromSchema(schema, { connection: -1 })).toThrowError(
+                'negative'
+            );
+            expect(() => buildTypeWeightsFromSchema(schema, { scalar: -1 })).toThrowError(
+                'negative'
+            );
+        });
+    });
+});
```

Comment:
It's probably worth adding a test for `mutation` and `subscription`. If not addressed at this point, then let's at least add a `// TODO: ` or create an issue so we come back to it.